WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -2051,7 +2051,26 @@ - (void)editorDidPressSettings:(WPEditorViewController *)editorController
 
 - (void)editorDidPressMedia:(WPEditorViewController *)editorController
 {
-    [self showMediaPicker];
+    if (self.post.blog.canUploadFiles) {
+        [self showMediaPicker];
+        return;
+    }
+
+    // first button of the toolbar
+    UIBarButtonItem *mediaButton = [editorController.toolbarView.items objectAtIndex:0];
+    UIView *buttonView = mediaButton.customView;
+    CGRect buttonFrame = [self.view convertRect:buttonView.frame
+                                       fromView:editorController.toolbarView];
+    CGRect targetFrame = CGRectMake(buttonFrame.origin.x,
+                                    buttonFrame.origin.y,
+                                    buttonFrame.size.width,
+                                    0.0);
+    NSString *tooltipText = NSLocalizedString(@"You are not allowed to add media on this blog",
```

Comment:
I concur.  Maybe something that just says, "Unable to add media to this site because of your Contributor role." — maybe not exactly that, but something that spells out this issue. Do we know if there are other situations where a user would not be able to upload media (besides the contrib role thing)?